sklearn/cross_decomposition/pls_.py

```diff
@@ -236,8 +236,10 @@ def fit(self, X, Y):
 
         # copy since this will contains the residuals (deflated) matrices
         check_consistent_length(X, Y)
-        X = check_array(X, dtype=np.float64, copy=self.copy)
-        Y = check_array(Y, dtype=np.float64, copy=self.copy, ensure_2d=False)
+        Xcopy = not X.flags.writable or self.copy
+        Ycopy = not Y.flags.writable or self.copy
```

Comment:
this should maybe factored into check_array:
Parameters copy=False|True|'on_readonly'